demos/notebooks/dqx_demo_library.ipynb

```diff
@@ -0,0 +1,272 @@
+    "                \"arguments\": {},\n",
+    "            },\n",
+    "            \"user_metadata\": {\"check_type\": \"completeness\", \"check_owner\": \"someone@email.com\"},\n",
+    "        },\n",
+    "        {\n",
+    "            \"criticality\": \"error\",\n",
+    "            \"check\": {\n",
+    "              \"function\": \"is_in_range\", \n",
+    "              \"for_each_column\": [\"age\"],\n",
+    "              \"arguments\": {\n",
+    "                \"min_limit\": 18,\n",
+    "                \"max_limit\": 120\n",
+    "                }\n",
+    "            },\n",
+    "        },\n",
+    "        {\n",
+    "            \"criticality\": \"error\",\n",
+    "            \"check\": {\n",
+    "              \"function\": \"is_in_list\", \n",
+    "              \"for_each_column\": [\"country\"],\n",
+    "              \"arguments\": {\n",
+    "                \"allowed\": [\"Germany\", \"France\"]\n",
+    "                }\n",
+    "            },\n",
+    "        },\n",
+    "    ]\n",
+    "\n",
+    "# Validate YAML checks\n",
+    "status = DQEngine.validate_checks(checks_from_object)\n",
+    "print(f\"Checks from Object: {status}\")\n",
+    "\n",
+    "checks_from_yaml = yaml.safe_load(\"\"\"\n",
+    "- check:\n",
+    "    function: is_not_null_and_not_empty\n",
+    "    for_each_column:\n",
+    "      - id\n",
+    "      - age\n",
+    "      - country\n",
+    "    criticality: error\n",
+    "- check:\n",
+    "    function: is_in_range\n",
+    "    for_each_column:\n",
+    "      - age\n",
+    "    criticality: warn\n",
+    "    arguments:\n",
+    "      min_limit: 18\n",
+    "      max_limit: 120\n",
+    "- check:\n",
+    "    function: is_in_list\n",
+    "    for_each_column:\n",
+    "      - country\n",
+    "    criticality: warn\n",
+    "    arguments:\n",
+    "      allowed:\n",
+    "        - Germany\n",
+    "        - France\n",
+    "\"\"\")\n",
+    "\n",
+    "# Validate YAML checks\n",
+    "status = DQEngine.validate_checks(checks_from_yaml)\n",
+    "print(f\"Checks from YAML: {status}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "vscode": {
+     "languageId": "polyglot-notebook"
+    }
+   },
+   "source": [
+    "### Setup `DQEngine`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ws = WorkspaceClient()\n",
+    "dq_engine = DQEngine(ws)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Apply Object Rules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "object_rules = DQEngineCore.build_checks_by_metadata(checks_from_object)\n",
+    "object_rules_valid_rows_df, object_rules_invalid_rows_df = dq_engine.apply_checks_and_split(new_users_df, object_rules)"
```

Comment:
Should also showcase usage of `apply_checks` without quarantining.

Reply:
Tested the `apply_checks` function, It throws an error, i don't know if i a missing something. But `apply_checks_by_metadata` seems to work.

<img width="1241" alt="image" src="https://github.com/user-attachments/assets/a0e182f0-9247-4cc1-a3dd-e8c81e3d6264" />

Reply:
apply_checks only works with checks defined using classes